extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/vertx/HttpInstrumenterVertxTracer.java

```diff
@@ -154,16 +155,18 @@ private static class ServerAttributesExtractor implements HttpServerAttributesGe
         @Override
         public String flavor(final HttpRequest request) {
             if (request instanceof HttpServerRequest) {
-                HttpServerRequest serverRequest = (HttpServerRequest) request;
-                switch (serverRequest.version()) {
-                    case HTTP_1_0:
-                        return "1.0";
-                    case HTTP_1_1:
-                        return "1.1";
-                    case HTTP_2:
-                        return "2.0";
-                    default:
-                        return null;
+                HttpVersion version = ((HttpServerRequest) request).version();
+                if (version != null) {
+                    switch (version) {
+                        case HTTP_1_0:
+                            return "1.0";
+                        case HTTP_1_1:
+                            return "1.1";
+                        case HTTP_2:
+                            return "2.0";
```

Comment:
This looks a bit fragile. What happens if they add 2.1 at some point?

Reply:
Yeah, not the most extensible thing. This happens to comply with the HTTP version semantics required by OTel, here: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#common-attributes
This gets the version from Vert.x... Not sure of the best way to proceed here.

Reply:
Yes, to follow the OTel naming conventions we need to convert the HTTP version format coming from Vert.x. We can probably be a bit smarter and use the value from Vert.x when we don't have a conversion available.

Reply:
sounds good to me, will update